pkg/sentry/sentry.go

```diff
@@ -0,0 +1,61 @@
+package sentry
+
+import (
+	"context"
+
+	"github.com/dapr/dapr/pkg/sentry/ca"
+	"github.com/dapr/dapr/pkg/sentry/config"
+	"github.com/dapr/dapr/pkg/sentry/server"
+	log "github.com/sirupsen/logrus"
+)
+
+type CertificateAuthority interface {
+	Run(context.Context, config.SentryConfig)
+}
+
+type sentry struct {
+	server server.CAServer
+}
+
+// NewSentryCA returns a new Sentry Certificate Authority instance.
+func NewSentryCA() CertificateAuthority {
+	return &sentry{}
+}
+
+// Run loads the trust anchors and issuer certs, creates a new CA and runs the CA server.
+func (s *sentry) Run(ctx context.Context, conf config.SentryConfig) {
+	// Create CA
+	certAuth, err := ca.NewCertificateAuthority(conf)
+	if err != nil {
+		log.Fatalf("error getting certificate authority: %s", err)
+	}
+	log.Info("certificate authority loaded")
+
+	// Load the trust
+	err = certAuth.LoadOrStoreTrustBundle()
```

Comment:
during cert rotation, some dapr runtimes would talk to ca server with old root cert. how do we resolve this issue? Also if the root cert is expired while running service, we should load new cert. Are we planning to redeploy ca service?

Reply:
Dapr root certs are valid for 10 years when issued by Dapr. Cert rotation is for workload certs, not root certs.

Root cert rotation will be done by the users, followed by restarting the sentry CA and the pods.

Reply:
10 years seems too long even between ca and dapr runtime internally.(I know this number comes from istio. I am curious about how istio rotates root without downtime.)

For the first iteration, it will be fine.

However, I am not sure how many users would use 10-year cert. If I work in azure production team, azure security team may concern about it and I would rotate this cert every month. Production team cannot restart service pod in production with downtime.

@amanbha @AaronCrawfis any thought?

Reply:
the lifetime of the root cert will be configurable.
Also, users can provide their own root certs if they want to.

There's no downtime of the entire system. This is also true for how Istio and Linkerd do it.

1. You replace root cert
2. Restart CA
3. Do a rolling update for the pods, so that you get a gradual rollout of the new root cert.

Reply:
I tried to learn more about how istio rotates the root cert. basically, they recommend to rotate every year even if the cert expiry is 10 years. 

https://istio.io/blog/2019/root-transition/
```
We strongly recommend you rotate root keys and root certificates annually as a security best practice. We will send out instructions for root key/cert rotation soon.
```

https://istio.io/docs/ops/configuration/security/root-transition/
Istio may have a way to keep using old cert to get new workload cert while rotation.

```If your Pilot does not have an Envoy sidecar, consider installing one. Pilot has issues using the old root certificate to verify the new workload certificates, which may cause disconnection between Pilot and Envoy. Please see here for how to check for this condition. The Istio upgrade guide by default installs Pilot with a sidecar.
```

As we discussed today, In production, couple of seconds or hundreds of millisecond causes the critical incident to the customers. we need to keep in mind for this case even if we will work on zero-downtime root rotation.

Reply:
Yep, we don't assume or take any special action about when the root cert gets rotated. For all we know, it could happen daily.

Reply:
@youngbupark, if the new root cert is signed with the same private key of the old root cert, there will be no downtime as older pods will be able to authenticate.

Reply:
> @youngbupark, if the new root cert is signed with the same private key of the old root cert, there will be no downtime as older pods will be able to authenticate.

ok good. we need to write the document. btw, if we can use crd to update the cert without restarting / redeploying pods, that would be nice.

Reply:
Actually, when you update a secret in k8s that is mounted on a pod, Kubernetes doesn't restart the pod or even the container inside the pod. the data gets refreshed in the volume automagically.